schedule/app.js

```diff
@@ -220,203 +260,331 @@ export function scheduleApp (options = {}) {
-      checkfuture = false
+      isToday = false
     }
     const {
       periods,
       alternate,
       summer,
-      getPeriodName,
-      isSELF,
       date: { ano, mez, dia, weekday }
     } = getSchedule(d)
     const day = days[weekday]
-    innerHTML = `<h2 class="schedule-dayname">${day}</h2><h3 class="schedule-date"><a class="totally-not-a-link" href="?date=${`${ano}-${mez +
-      1}-${dia}`}">${localizeWith('date', 'times', {
-      M: months[mez],
-      D: dia
-    })}</a></h3>`
+    const isSchool = !summer && periods.length
+    const noSchool = !summer && !periods.length
     const assignments = options.getAssignments(d)
-    if (assignments.noPeriod) {
-      innerHTML += assignments.noPeriod
-    }
-    if (summer)
-      return (
-        innerHTML +
-        `<span class="schedule-noschool">${localize('summer')}</span>`
-      )
-    if (alternate) {
-      innerHTML += `<span class="schedule-alternatemsg">${localizeWith(
-        'alt-msg',
-        'other',
-        { D: `<strong>${alternate.description}</strong>` }
-      )}</span>`
-    }
-    if (periods.length) {
-      // If a day ends in an optional period, don't count it.
-      // 'Flex' is temporarily in the list because it's kind of optional this
-      // year.
-      const optionalPeriods = ['Lunch', 'Brunch', 'Flex']
+
+    const optionalPeriods = ['Lunch', 'Brunch', 'Flex']
+    let schedule = []
+    if (isSchool) {
+      schedule = []
+
       const [lastRequiredPeriod] = periods
         .filter(({ name }) => !optionalPeriods.includes(name))
         .slice(-1)
       if (lastRequiredPeriod) {
-        innerHTML += `<span class="schedule-end">${localizeWith(
-          'end-time',
-          'times',
-          {
-            T: `<strong>${getHumanTime(
-              ('0' + lastRequiredPeriod.end.hour).slice(-2) +
-                ('0' + lastRequiredPeriod.end.minute).slice(-2)
-            )}</strong>`
-          }
-        )}</span>`
+        schedule.push([
+          'span.schedule-end',
+          applyEndTime({
+            T: getHumanTime(lastRequiredPeriod.end)
+          })
+        ])
       }
-      // QUESTION: Should there be feedback for days with only optional periods?
-      // Later QUESTION: What did I mean by "feedback"??
-      // "Feedback" as in a note that the entire day is optional.
-      if (checkfuture) {
-        let i
-        for (i = 0; i < periods.length; i++)
-          if (totalminute < periods[i].end.totalminutes) break
-        let str
-        let compactTime, period, compactStr
-        if (i >= periods.length) {
-          str = `<p class="schedule-endingin">${localizeWith('ended', 'times', {
-            P: getPeriodSpan((period = getPeriodName(periods.length - 1))),
-            T: `<strong>${(compactTime = getUsefulTimePhrase(
-              totalminute - periods[periods.length - 1].end.totalminutes
-            ))}</strong>`
-          })}</p>`
-          compactStr = localize('appname')
```

Comment:
So apparently UGWA doesn't show the app name after school if not using compact durations